libs/community/langchain_community/document_loaders/sharepoint.py

```diff
@@ -90,38 +117,58 @@ def lazy_load(self) -> Iterator[Document]:
                     blob_part.metadata.update(blob.metadata)
                     yield blob_part
 
-    def authorized_identities(self) -> List:
+    def authorized_identities(self, file_id: str) -> List:
         data = self._fetch_access_token()
         access_token = data.get("access_token")
         url = (
-            f"https://graph.microsoft.com/v1.0/sites/{self.site_id}/"
-            f"drives/{self.document_library_id}/items/{self.file_id}/permissions"
+            "https://graph.microsoft.com/v1.0/drives"
+            f"/{self.document_library_id}/items/{file_id}/permissions"
         )
         headers = {"Authorization": f"Bearer {access_token}"}
-        response = requests.request("GET", url, headers=headers, data={})
-        groups_list = response.json()
+        response = requests.request("GET", url, headers=headers)
+        access_list = response.json()
 
         group_names = []
 
-        for group_data in groups_list.get("value"):
-            if group_data.get("grantedToV2"):
-                if group_data.get("grantedToV2").get("siteGroup"):
-                    site_data = group_data.get("grantedToV2").get("siteGroup")
-                    # print(group_data)
-                    group_names.append(site_data.get("displayName"))
-                elif group_data.get("grantedToV2").get("group") or (
-                    group_data.get("grantedToV2").get("user")
-                ):
-                    site_data = group_data.get("grantedToV2").get("group") or (
-                        group_data.get("grantedToV2").get("user")
-                    )
-                    # print(group_data)
-                    group_names.append(site_data.get("displayName"))
-
+        for access_data in access_list.get("value"):
+            if access_data.get("grantedToV2"):
+                site_data = (
+                    (access_data.get("grantedToV2").get("siteUser"))
+                    or (access_data.get("grantedToV2").get("user"))
+                    or (access_data.get("grantedToV2").get("group"))
+                )
+                if site_data:
+                    email = site_data.get("email")
+                    if email:
+                        group_names.append(email)
         return group_names
 
     def _fetch_access_token(self) -> Any:
         with open(self.token_path) as f:
             s = f.read()
         data = json.loads(s)
         return data
+
+    def get_extended_metadata(self, file_id: str) -> dict:
```

Comment:
I am new to Sharepoint, so I don't know too much about it's usability - why is this function helpful/will be used by users?